pkg/server/server.go

```
@@ -1,56 +1,106 @@
+// Start listens on the configured port and serves until ctx is cancelled.
 //
-// Returns an error if the server fails to start or encounters a runtime error.
-func Start(port int, logger zerolog.Logger) error {
-	addr := fmt.Sprintf(":%d", port)
+// It returns nil after a clean shutdown. A listen failure, a serve failure, or
+// a shutdown that exceeds shutdownTimeout each return an error. By the time
+// Start returns, the serve goroutine has exited: a caller that saw Start return
+// knows nothing is still running, which is what makes a leak a test failure
+// rather than a suspicion.
+func (s *Server) Start(ctx context.Context) error {
+	ln, err := net.Listen("tcp", fmt.Sprintf(":%d", s.port))
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)